doc/internal/man3/ossl_rand_get_entropy.pod

```diff
@@ -41,18 +52,34 @@ The output is stored in a buffer which contains at least I<min_len> and at
 most I<max_len> bytes.  The buffer address is stored in I<*pout> and the
 buffer length returned to the caller.
 
+ossl_rand_get_user_nonce() is the same as ossl_rand_get_nonce() except
+that it retrieves the seeding material from the library context's DRBG
+seed source.  By default this is the operating system but it can be
+changed by calling L<RAND_set_seed_source_type(3)>.
+
 ossl_rand_cleanup_nonce() cleanses and frees any storage allocated by
-ossl_rand_get_nonce().  The nonce buffer is pointed to by I<buf> and is
-of length I<len> bytes.
+ossl_rand_get_nonce() or ossl_rand_get_user_nonce().  The nonce buffer
+is pointed to by I<buf> and is of length I<len> bytes.
+
+=head1 NOTES
+
+FIPS providers 3.0.0, 3.0.8 and 3.0.9 incorrectly pass a provider
+internal pointer to ossl_rand_get_entropy(), ossl_rand_cleanup_entropy(),
+ossl_rand_get_nonce() and ossl_rand_cleanup_nonce().  This pointer cannot
+be safely dereferenced.
 
 =head1 RETURN VALUES
 
-ossl_rand_get_entropy() and ossl_rand_get_nonce() return the number of bytes
-in I<*pout> or 0 on error.
+ossl_rand_get_entropy(), ossl_rand_get_user_entropy(),
+ossl_rand_get_nonce() and ossl_rand_get_user_nonce() return the number
+of bytes in I<*pout> or 0 on error.
 
 =head1 HISTORY
 
-The functions described here were all added in OpenSSL 3.0.
+The functions ossl_rand_get_user_entropy() and ossl_rand_get_user_nonce()
+were added in OpenSSL 3.0.12, 3.1.4 and 3.2.0.
```

Comment:
### Urgent

@arapov @openssl/otc:  this HISTORY entry claims that the fix not only went into master and openssl-3.1, but also into openssl-3.0. It seems like that was Pauli's original intent, as indicated by branch labels on this pull request. However, the fix has apparently only been merged to master (this pr) and to openssl-3.1 (#22210).

Related to my [AOB question from today's OTC meeting](https://github.com/openssl/otc/commit/4d78ff0dbc6a187ba55f72ca6fe6adc829cc6f72#diff-908dacd1ccb46af98978eda22646a8f950b5294f9bec41b7854f35d3f1ad0e53R113-R117).

Reply:
I found this note by @paulidale mentioning problems with backporting to 3.0: https://github.com/openssl/openssl/pull/22210#issuecomment-1738292407.